pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -28,6 +28,43 @@
 INV_SQRT2 = 1 / qml.math.sqrt(2)
 
 
+class Barrier(Operation):
+    r"""Barrier(wires)
+    The Barrier operator, used to separate the compilation process into blocks or as a visual tool.
+
+    **Details:**
+
+    * Number of wires: AnyWires
+    * Number of parameters: 0
+
+    Args:
+        only_visual (bool): True if we do not want it to have an impact on the compilation process. Default is False.
+        wires (Sequence[int] or int): the wires the operation acts on
+    """
+    num_params = 0
+    num_wires = AnyWires
+    par_domain = None
+
+    def __init__(self, only_visual=False, *params, wires=None, do_queue=True, id=None):
```

Comment:
Since it will never have parameters, we can probably remove `*params` from the call signature.